src/Ninject.Web.AspNetCore/NinjectServiceProviderBuilder.cs

```diff
@@ -18,7 +22,10 @@ public NinjectServiceProvider Build()
 		{
 			var result = new NinjectServiceProvider(_kernel);
 			_kernel.Bind<IServiceProvider>().ToConstant(result); // needed for factory methods in IServiceCollection
-			_kernel.Populate(_services);
+
+			var adapter = new ServiceCollectionAdapter();
```

Comment:
Yes, it's better to separate this functionality from the kernel and not use an extension method 👍